common/serial/portio_linux.h

```diff
@@ -0,0 +1,28 @@
+#ifndef PORTIO_LINUX__H
+#define PORTIO_LINUX__H
+
+#include <port_io.h>
+#include <stdio.h>
+
+extern void bf_outb(uint16_t port, uint8_t value);
+extern void bf_outw(uint16_t port, uint16_t value);
+extern uint8_t bf_inb(uint16_t port);
+extern uint16_t bf_inw(uint16_t port);
+
+#define TRACE() printf("%s:%d\n", __PRETTY_FUNCTION__, __LINE__)
+#define TRACE_INT(x) printf("[%s:%d] - %08X\n", __PRETTY_FUNCTION__, __LINE__, x)
+
+class portio_linux : public port_io {
+public:
+
+	portio_linux();
```

Comment:
the way I have been doing this is:

```
include \
    blah.h
    blah_base.h

src \
    arch \
        linux \
            blah.cpp
        vmm \
            blah.cpp
```

In the Makefile, you will add 
LINUX_SOURCES+= blah.cpp

The Cross-Compiled part has not been added to the makefiles yet (will do that soon), but it will likely end up with a 
VMM_SOURCES+=blah.cpp

Then add:
VPATH+=arch/linux arch/vmm

In your unit test, you will use blah_base in the hippomock.